File: src/app/model/produto.ts
export class Produto{
    public id: number; 
    public nome: string; 
    public descricao: string;
    public des_detalhada: string;
    public preco: number; 
    public img: string; 
    public quant: number;

    constructor(id: number, nome: string, descricao: string, des_detalhada: string, preco: number, img: string, quant: number) {
        this.id = id;
        this.nome = nome;
        this.descricao = descricao;
        this.des_detalhada = des_detalhada;
        this.preco = preco;
        this.img = img;
        this.quant = quant;
    }
}

File: Back/sci-fi_worlds/src/main/java/com/web/sci_fi_worlds/app/entity/Produto.java
package com.web.sci_fi_worlds.app.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Produto {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO) 
    private int codigo;
    private String nome;
    private String descricao;
    private double preco;
    private int estoque;
    private String des_detalhada;
    private String img;
    private int destaque;
    
    public int getCodigo() {
        return codigo;
    }
    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }
    public String getNome() {
        return nome;
    }
    public void setNome(String nome) {
        this.nome = nome;
    }
    public String getDescricao() {
        return descricao;
    }
    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }
    public double getPreco() {
        return preco;
    }
    public void setPreco(double preco) {
        this.preco = preco;
    }
    public int getestoque() {
        return estoque;
    }
    public void setestoque(int estoque) {
        this.estoque = estoque;
    }
        public String getDes_detalhada() {
        return des_detalhada;
    }
    public void setDes_detalhada(String des_detalhada) {
        this.des_detalhada = des_detalhada;
    }
        public String getImg() {
        return img;
    }
    public void setImg(String img) {
        this.img = img;
    }
        public int getDestaque() {
        return destaque;
    }
    public void setDestaque(int destaque) {
        this.destaque = destaque;
    }
}
